### PR TITLE
[BUG] Catch retry submit exception

### LINKED ISSUE #4795: [BUG] retry submit exception is not catched
**Describe the bug**
When `Load Job Task Queue` is filled, continue to submit more jobs to the queue will cause `RejectedExecutionException`, and the load job will be retried. But `callback.onTaskFailed` function does not catch the exception, that will cause re-submitting job failed, and status is not updated to failed.

**To Reproduce**
1. set FE configuration: async_load_task_pool_size=30
2. create a table
3. create more than 60 broker load jobs, start them concurrently to fill the broker load job queue.
2020-10-26 19-46-31屏幕截图
2020-10-26 19-46-49屏幕截图
the status of job is not updated to failed.
2020-10-26 21-48-50屏幕截图

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by xinghuayu007**
## Proposed changes

When `Load Job Task Queue` is filled, continue to submit more jobs to the queue will cause `RejectedExecutionException`. But `callback.onTaskFailed` function does not catch the exception, that will cause re-submitting job failed, and status is not updated to failed.
issue: #4795 
## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on (Fix #4795), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If this change need a document change, I have updated the document
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/load/loadv2/BulkLoadJob.java [hunk 1]
```diff
@@ -233,7 +233,32 @@ public void onTaskFailed(long taskId, FailMsg failMsg) {
...
+                        Catalog.getCurrentCatalog().getLoadTaskScheduler().submit(loadTask);
+                        retrySuccessFlag = true;
+                        break;
+                    } catch (Exception e) {
+                        try {
+                            Thread.sleep(100);
+                        } catch (InterruptedException ex) {
+                            LOG.warn("retry sublit task sleep failed. task_id:{}", id);
+                        }
+                        // retry task
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/org/apache/doris/load/loadv2/BulkLoadJob.java:248]: The code is redundant:
```
                        idToTasks.remove(loadTask.getSignature());
                        if (loadTask instanceof LoadLoadingTask) {
                            loadStatistic.removeLoad(((LoadLoadingTask) loadTask).getLoadId());
                        }
                        loadTask.updateRetryInfo();
                        idToTasks.put(loadTask.getSignature(), loadTask);

```

Comment (xinghuayu007) [reply fe/fe-core/src/main/java/org/apache/doris/load/loadv2/BulkLoadJob.java]: ＯＫ，Ｉmodified it.

Comment (xinghuayu007) [reply fe/fe-core/src/main/java/org/apache/doris/load/loadv2/BulkLoadJob.java]: > The code is redundant:
> 
> ```
>                         idToTasks.remove(loadTask.getSignature());
>                         if (loadTask instanceof LoadLoadingTask) {
>                             loadStatistic.removeLoad(((LoadLoadingTask) loadTask).getLoadId());
>                         }
>                         loadTask.updateRetryInfo();
>                         idToTasks.put(loadTask.getSignature(), loadTask);
> ```

Maybe, it is better to catch the exception in `fe-core/src/main/java/org/apache/doris/task/MasterTaskExecutor.java#submit` and return false. But currently, too many calls refer this function, may need too much modfication.

**Review by morningman — COMMENTED** [review]
Some other parts need to be fixed too:

image

**Review by EmmyMiao87 — CHANGES_REQUESTED** [review]
Another problem is that I think loading should not have so much concurrency control.
In fact, db itself already has a control of the number of transactions.
So the number of concurrent tasks here is best to be twice the number of transactions.

Also, if the submitted task is abnormal, I think the load job can be cancelled directly.
The reason is that after the system is currently loading the stack, it is generally difficult to recover quickly, so you are not sure how long you will have to wait to retry successfully.

**Review by morningman — APPROVED** [review]
LGTM

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**